src/ec/curve25519/ed25519.rs

```diff
@@ -147,9 +144,8 @@ impl<'a> Ed25519KeyPair {
             let mut r = ExtPoint::new_at_infinity();
             unsafe {
                 GFp_x25519_ge_scalarmult_base(&mut r, &nonce);
-                GFp_ge_p3_tobytes(signature_r, &r);
             }
-
+            signature_r.copy_from_slice(&r.encode());
```

Comment:
We don't need to worry about performance changes regarding adding a few *one-time* 32-byte or 64-byte copies. The things that are likely to significantly affect performance are: 
 
1. Making some C function `extern` that was `static` in a way that causes the C compiler to optimize it (and probably the whole curve25519.c file) worse.

2. When passing arrays from one function to another, when we change the calls from C -> C to Rust -> C, worse code could get generated because this would inhibit autovectorization and/or inhibit the compiler's ability to rely on arrays being aligned and/or similar things.